source/Calamari.Tests/Fixtures/PackageRetention/JournalFixture.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Threading.Tasks;
+using Calamari.Common.Features.Processes.Semaphores;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.PackageRetention.Model;
+using Calamari.Deployment.PackageRetention.Repositories;
+using Calamari.Tests.Fixtures.Integration.FileSystem;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using Newtonsoft.Json;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.PackageRetention
+{
+    [TestFixture]
+    public class JournalFixture
+    {
+        static readonly string TentacleHome = TestEnvironment.GetTestPath("Fixtures", "Journal");
+
+        [SetUp]
+        public void SetUp()
+        {
+            if (!Directory.Exists(TentacleHome))
+                Directory.CreateDirectory(TentacleHome);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (Directory.Exists(TentacleHome))
+                Directory.Delete(TentacleHome, true);
+        }
+
+        [Test]
+        public void WhenMultipleThreadsTryToWriteToTheJournalAtTheSameTime_ThenTheWritesAreHandledCorrectly()
+        {
+            //TODO: Do this using Autofac
+            TypeDescriptor.AddAttributes(typeof(ServerTaskId), new TypeConverterAttribute(typeof(TinyTypeTypeConverter<ServerTaskId>)));
```

Comment:
The test was failing on converting `ServerTaskId` so I copied this over from `Program` for now.